MoreLinq/MoreEnumerable.cs

```diff
@@ -27,5 +27,18 @@ namespace MoreLinq
     
     public static partial class MoreEnumerable
     {
+        /// <summary>
+        /// Gets the count of an enumerable if it is precalculated in <see cref="ICollection{T}"/>
+        /// or <see cref="IReadOnlyCollection{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of objects to enumerate.</typeparam>
+        /// <param name="source">The enumerable to get the count for.</param>
+        /// <returns>The count if it is available without enumeration, or null if it is not.</returns>
+        static int? TryGetCollectionCount<T>(this IEnumerable<T> source)
```

Comment:
Shouldn't we be validating that the `source` argument is not `null`?

Reply:
Yes.